app/services/members/create_service.rb

```diff
@@ -24,11 +25,30 @@ def execute # rubocop:disable Metrics/AbcSize
                                         namespace_type: namespace.class.model_name.human)
       end
 
-      member.save
+      had_access = Member.can_view?(member.user, namespace, true) if member.valid?
```

Comment:
Can we change `had_access` to `has_previous_access` and then only call send emails if the member doesn't have previous access. No need to go into the method and return right away if they have previous access

Reply:
Please see 9fae2990a556922b786a2f5a4114846bcfb0f33f.